sdks/python/apache_beam/io/avroio_test.py

```diff
@@ -472,10 +476,55 @@ def test_sink_transform_snappy(self):
         assert_that(readback, equal_to([json.dumps(r) for r in self.RECORDS]))
 
 
-class TestFastAvro(TestAvro):
+@unittest.skipIf(sys.version_info[0] == 3 and
+                 os.environ.get('RUN_SKIPPED_PY3_TESTS') != '1',
+                 'This test still needs to be fixed on Python 3. '
+                 'TODO: BEAM-6522.')
+class TestAvro(AvroBase, unittest.TestCase):
+  def __init__(self, methodName='runTest'):
+    super(TestAvro, self).__init__(methodName)
+    self.use_fastavro = False
+    self.SCHEMA = Parse(self.SCHEMA_STRING)
+
+  def _write_data(self,
+                  directory=None,
+                  prefix=tempfile.template,
+                  codec='null',
+                  count=len(RECORDS)):
+    with tempfile.NamedTemporaryFile(delete=False,
+                                     dir=directory,
+                                     prefix=prefix) as f:
+      writer = DataFileWriter(f, DatumWriter(), self.SCHEMA, codec=codec)
+      len_records = len(self.RECORDS)
+      for i in range(count):
+        writer.append(self.RECORDS[i % len_records])
+      writer.close()
+      self._temp_files.append(f.name)
+      return f.name
+
+
+class TestFastAvro(AvroBase, unittest.TestCase):
   def __init__(self, methodName='runTest'):
     super(TestFastAvro, self).__init__(methodName)
     self.use_fastavro = True
+    self.SCHEMA = parse_schema(json.loads(self.SCHEMA_STRING))
+
+  def _write_data(self,
+                  directory=None,
+                  prefix=tempfile.template,
+                  codec='null',
+                  count=len(RECORDS),
+                  sync_interval=16000):
```

Comment:
Let's not redefine default values for `writer()` method in the test. Currently the defaults for `codec` and `sync_interval` match the respective defaults in `fastavro/avro`, but if the library implementation changes the default in the future, the test may be still hardcoding old value. How about something like:
```
  def _write_data(self,
                  directory=None,
                  prefix=tempfile.template,
                  count=len(RECORDS),
                  **kwargs):
    ...
    writer(f, self.SCHEMA, all_records, **kwargs)
```